pilot/pkg/networking/core/v1alpha3/accesslog.go

```diff
@@ -167,28 +167,33 @@ func (b *AccessLogBuilder) setTCPAccessLog(push *model.PushContext, proxy *model
 		return
 	}
 
-	if al := buildAccessLogFromTelemetry(push, cfg, false); len(al) != 0 {
+	if al := buildAccessLogFromTelemetry(push, cfgs, false); len(al) != 0 {
 		tcp.AccessLog = append(tcp.AccessLog, al...)
 	}
 }
 
-func buildAccessLogFromTelemetry(push *model.PushContext, spec *model.LoggingConfig, forListener bool) []*accesslog.AccessLog {
+func buildAccessLogFromTelemetry(push *model.PushContext, cfgs []*model.LoggingConfig, forListener bool) []*accesslog.AccessLog {
 	als := make([]*accesslog.AccessLog, 0)
-	telFilter := buildAccessLogFilterFromTelemetry(spec)
-	filters := []*accesslog.AccessLogFilter{}
-	if forListener {
-		filters = append(filters, addAccessLogFilter())
-	}
-	if telFilter != nil {
-		filters = append(filters, telFilter)
-	}
+	for _, spec := range cfgs {
+		telFilter := buildAccessLogFilterFromTelemetry(spec)
+		filters := []*accesslog.AccessLogFilter{}
+		if forListener {
+			filters = append(filters, addAccessLogFilter())
+		}
+		if telFilter != nil {
+			filters = append(filters, telFilter)
+		}
+
+		if spec.Provider == nil {
+			// should not happen, just in case
+			continue
+		}
 
-	for _, p := range spec.Providers {
 		var al *accesslog.AccessLog
-		switch prov := p.Provider.(type) {
+		switch prov := spec.Provider.Provider.(type) {
 		case *meshconfig.MeshConfig_ExtensionProvider_EnvoyFileAccessLog:
 			// For built-in provider, fallback to Mesh Config for formatting options.
-			if p.Name == defaultEnvoyAccessLogProvider {
```

Comment:
we have discuss this in https://github.com/istio/istio/pull/39606

Reply:
I don't understand.  Can you please clarify?  That PR was about falling back to MeshConfig. This logic is replacing the provider completely. That will break user expectations completely.

Reply:
I got what your said, do you means, we should only fallback to MeshConfig when `p.Name == "envoy"` and `prov.EnvoyFileAccessLog.Path==""`?

Reply:
let's leave this for now. i'm worried that people will create their own provider named `envoy` and this will be a mess, but we can address that later.